### PR TITLE
test: Migrate devices/views/details to RTL

### PR DESCRIPTION
**Pull request by ndv99**
## Done

Migrated the following files to RTL:
- [src/app/dashboard/views/DiscoveryAddForm/DiscoveryAddForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-c69888ce48fc3fbddcabe3d0de024b340fbed40da093dc626641083f381d460d)
- [src/app/devices/views/DeviceDetails/DeviceDetails.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-edc95ebb0d15ec84731b5a8aad506aef543ff92f543ab55647bcc6e4c5e8ad3c)
- [src/app/devices/views/DeviceDetails/DeviceDetailsHeader/DeviceDetailsHeader.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-4ecb3ca2b8a034d93d69ac1d9da34cc8855d591482184676cdc6ae57bc27dd54)
- [src/app/devices/views/DeviceDetails/DeviceDetailsHeader/DeviceName/DeviceName.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-6f94237d6c699c335b66529d21a97fdc6068f8695140e2474d8814586ad3ccd2)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/AddInterface/AddInterface.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-d68dfafe660734a5f4a0264514b159204317a9b8fde15cfd54a2b19445d69fb2)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/DeviceNetwork.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-6b5918bca4733c87293558f7959130f90fa74dc72e78c09bdfb6864ba53be001)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/DeviceNetworkTable/DeviceNetworkTable.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-e9d53a786ea9c48674a8477eb2abfaacda9fb902063c37f11064d0d795701753)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/DeviceNetworkTable/RemoveInterface/RemoveInterface.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-bf3a50b28eaf37afc44f6f8d49fdd54433bac98b2a0dce5b1b9992836da6837d)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/EditInterface/EditInterface.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-4f6c492e9fef4a7c30903113eefe402d0e1b972975443cbef4afd3471353a674)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/EditInterface/EditInterfaceTable/EditInterfaceTable.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-b4c7bc0a8ef3c042d097f32aef7756d0a705bdd1ef62c125e782923a5657d70c)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/InterfaceForm/InterfaceForm.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-a7caa4c432bd5ea3a80a70fb06f2ecf616e2d448b824301203dd6d614afdbf91)
- [src/app/devices/views/DeviceDetails/DeviceNetwork/InterfaceForm/InterfaceFormFields/InterfaceFormFields.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-a83581f8d50060814da34a51cd2deeeb7e43e11e42e58dfa524c1fb58a295fd0)
- [src/app/devices/views/DeviceDetails/DeviceSummary/DeviceOverviewCard/DeviceOverviewCard.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-f741e58a98cf15ffa9fb7ff91f8b6abe2f18b9714fe18b19ba24506780cbfa06)
- [src/app/devices/views/DeviceDetails/DeviceSummary/DeviceSummary.test.tsx](https://github.com/canonical/maas-ui/compare/main...ndv99:test-migrate-device-views-details?expand=1#diff-490d07aec58da37199b9808969966c86da22d43b137c3418df2880effe446ec8)

### CODE REVIEW

**Comment by petermakowski** [comment]
Whoah! That's a lot of tests migrated @ndv99 🎉

**Comment by ndv99** [comment]
> Whoah! That's a lot of tests migrated @ ndv99 :tada: 

Thank you thank you :sunglasses:

File: src/app/devices/views/DeviceDetails/DeviceDetails.test.tsx [hunk 1]
```diff
@@ -154,25 +124,13 @@ describe("DeviceDetails", () => {
...
-      </Provider>
-    );
+    renderWithBrowserRouter(<DeviceDetails />, {
+      route: urls.devices.device.index({ id: device.system_id }),
+      store,
+      routePattern: `${urls.devices.device.index(null)}/*`,
+    });
 
-    expect(wrapper.find("[data-testid='not-found']").exists()).toBe(true);
+    // expect(wrapper.find("[data-testid='not-found']").exists()).toBe(true);
```
Comment (petermakowski) [inline src/app/devices/views/DeviceDetails/DeviceDetails.test.tsx:133]: Remove please

File: src/app/devices/views/DeviceDetails/DeviceDetailsHeader/DeviceDetailsHeader.test.tsx [hunk 2]
```diff
@@ -31,105 +29,87 @@ describe("DeviceDetailsHeader", () => {
...
+        systemId="abc123"
+      />,
+      { store }
     );
 
-    expect(wrapper.find("SectionHeader").prop("loading")).toBe(true);
-    expect(wrapper.find("SectionHeader").prop("subtitleLoading")).not.toBe(
-      true
-    );
+    expect(screen.getByRole("heading")).toHaveTextContent("Loading...");
```
Comment (petermakowski) [inline src/app/devices/views/DeviceDetails/DeviceDetailsHeader/DeviceDetailsHeader.test.tsx:41]: Might be better to stick to a single way of doing the same thing (you also use `screen.getByTestId("section-header-title")` in a test below).

File: src/app/devices/views/DeviceDetails/DeviceNetwork/DeviceNetwork.test.tsx [hunk 3]
```diff
@@ -44,19 +37,12 @@ describe("DeviceNetwork", () => {
...
-    );
-    expect(wrapper.find("NodeNetworkTab").exists()).toBe(true);
-    expect(wrapper.find("DHCPTable").exists()).toBe(true);
-    expect(wrapper.find("Spinner").exists()).toBe(false);
+    renderWithBrowserRouter(<DeviceNetwork systemId="abc123" />, {
+      route: "/device/abc123",
+      store,
+    });
+    expect(screen.getByLabelText("Device network")).toBeInTheDocument();
+    expect(screen.getAllByRole("grid")).toHaveLength(2);
```
Comment (petermakowski) [inline src/app/devices/views/DeviceDetails/DeviceNetwork/DeviceNetwork.test.tsx:45]: Without looking at the implementation - I'm assuming the length will be 1 when a specific name is used? 

```suggestion
    expect(screen.getAllByRole("grid", { name: /DHCP/ )).toBeInTheDocument();
```

File: src/app/devices/views/DeviceDetails/DeviceDetails.test.tsx (hunk 1 above)
Comment (ndv99) [reply src/app/devices/views/DeviceDetails/DeviceDetails.test.tsx:133]: :man_facepalming:

File: src/app/devices/views/DeviceDetails/DeviceNetwork/DeviceNetwork.test.tsx (hunk 3 above)
Comment (ndv99) [reply src/app/devices/views/DeviceDetails/DeviceNetwork/DeviceNetwork.test.tsx:45]: That works actually yeah! Only thing is the first table doesn't appear to have a name, so I just gave it an aria label

**Review by petermakowski — APPROVED** [review]

**Pull request merged**